source/slang/slang-check-modifier.cpp

```diff
@@ -2211,6 +2211,13 @@ void SemanticsVisitor::checkRayPayloadStructFields(StructDecl* structDecl)
         return;
     }
 
+    // Define valid stage names
+    HashSet<String> validStages;
+    validStages.add("anyhit");
+    validStages.add("closesthit");
+    validStages.add("miss");
+    validStages.add("caller");
```

Comment:
should be able to do `const HashSet<String> validStages("a", "b", "c");` I think